Framework/NexusGeometry/src/NexusGeometryParser.cpp

```diff
@@ -737,26 +632,25 @@ extractInstrument(const H5File &file, const Group &root) {
     Pixels detectorPixels = Eigen::Affine3d::Identity() * pixelOffsets;
     bool searchTubes = false;
     // Extract shape
-    auto shape = parseNexusShape(detectorGroup, searchTubes);
+    auto detShape = parseNexusShape(detectorGroup, searchTubes);
 
-    std::vector<Tube> tubes;
+    std::vector<detail::Tube> tubes;
     if (searchTubes) {
-      tubes = findTubes(*shape, pixelOffsets, detectorIds);
+      tubes = TubeHelpers::findTubes(*detShape, pixelOffsets, detectorIds);
```

Comment:
I don't know if you want to offload to the `builder` to create the whole set of tubes. i.e: `builder.addTubes(bankName, tubes);`